### PR TITLE
nfc: Add library with NDEF buffer change

### PR DESCRIPTION
**Pull request by KAGA164**
This commits provides the NDEF buffer change for
NFC T4T library. This feature is needed for NFC
TNEP Protocol.

Signed-off-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>

### CODE REVIEW

**Comment by KAGA164** [comment]
@grochu @kapi-no Please review

File: nfc/include/nfc_t4t_lib.h [hunk 1]
```diff
@@ -250,7 +255,7 @@ int nfc_t4t_response_pdu_send(const u8_t *pdu, size_t pdu_length);
 
 /** @brief Set an NFC parameter.
  *
- * Allows to set an NFC configuration parameter.
+ * Allows to set any parameter defined as available by HAL_NFC.
```
Comment (kapi-no) [inline nfc/include/nfc_t4t_lib.h]: Why is the description changed for this API? HAL_NFC is in the nrfx - I don't think this function allows to change any parameter in HAL_NFC

File: nfc/include/nfc_t4t_lib.h [hunk 2]
```diff
@@ -197,15 +198,19 @@ int nfc_t4t_setup(nfc_t4t_callback_t callback, void *context);
...
  *
  * @param emulation_buffer Buffer pointer
  * @param buffer_length Length of buffer (maximum writable NDEF size)
  *
  * @retval 0 Success.
  * @retval -EINVAL Invalid argument (e.g. wrong data length).
  * @retval -EINVAL Invalid argument (e.g. NULL pointer).
- * @retval -ENOTSUP If emulation is in running state.
+ * @retval -ENOTSUP If the new buffer has different length than the first one.
+ * @retval -EFAULT If provided buffer is the current used buffer.
```
Comment (kapi-no) [inline nfc/include/nfc_t4t_lib.h]:
```suggestion
 * @retval -EFAULT If the provided buffer is the currently used buffer.
```

File: nfc/include/nfc_t4t_lib.h [hunk 3]
```diff
@@ -197,15 +198,19 @@ int nfc_t4t_setup(nfc_t4t_callback_t callback, void *context);
...
+ * critical sections or disable the interrupts.
  *
  * @param emulation_buffer Buffer pointer
  * @param buffer_length Length of buffer (maximum writable NDEF size)
  *
  * @retval 0 Success.
  * @retval -EINVAL Invalid argument (e.g. wrong data length).
  * @retval -EINVAL Invalid argument (e.g. NULL pointer).
- * @retval -ENOTSUP If emulation is in running state.
+ * @retval -ENOTSUP If the new buffer has different length than the first one.
```
Comment (kapi-no) [inline nfc/include/nfc_t4t_lib.h]:
```suggestion
 * @retval -ENOTSUP If the new buffer has a different length than the first one.
```

File: nfc/include/nfc_t4t_lib.h (hunk 1 above)
Comment (KAGA164) [reply nfc/include/nfc_t4t_lib.h]: Honestly  I do not know. This had to be changed previously by somebody

Comment (kapi-no) [reply nfc/include/nfc_t4t_lib.h]: Hmm, then we need to determine which description is more up to date and sync nrfxlib with original docs

Comment (grochu) [reply nfc/include/nfc_t4t_lib.h]: I updated the dox in https://github.com/NordicPlayground/nrfxlib/pull/84, so you'd just need to rebase. Though manifest in fw-nrfconnect-nrf (https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1534) doesn't point to the updated dox in nrfxlib, as other manifest PRs were blocking the update.

File: nfc/include/nfc_t4t_lib.h [hunk 4]
```diff
@@ -197,15 +198,19 @@ int nfc_t4t_setup(nfc_t4t_callback_t callback, void *context);
  *
  * The buffer needs to be kept accessible for the lifetime of the emulation.
  * If an external Reader-Writer changes the NDEF content it is signaled through
- * the app-callback.
+ * the app-callback. Buffer can be changed during the lifetime of the emulation,
+ * when NDEF READ or UPDATE procedure is pending, and it will be changed after
+ * this procedure is finished. To perform this procedure safely use a
```
Comment (grochu) [inline nfc/include/nfc_t4t_lib.h]: use critical sections

Comment (KAGA164) [reply nfc/include/nfc_t4t_lib.h]: Ok

File: nfc/include/nfc_t4t_lib.h (hunk 3 above)
Comment (KAGA164) [reply nfc/include/nfc_t4t_lib.h]: Ok

File: nfc/include/nfc_t4t_lib.h (hunk 2 above)
Comment (KAGA164) [reply nfc/include/nfc_t4t_lib.h]: Ok

**Comment by grochu** [comment]
Please create a manifest PR in fw-nrfconnect-nrf

**Pull request merged**